File: frontend/src/api/jobs.ts
import { requestJson } from './client'

export interface Job {
  id: number
  title: string
  raw_text: string
  profile: Record<string, unknown>
  created_at: string
}

export interface CreateJobPayload {
  title: string
  raw_text: string
}

export interface CompareDimension {
  name: string
  category: string
  required_level: string
  weight: number
}

export interface CompareItem {
  job_id: number
  job_title: string
  dimensions: CompareDimension[]
}

export interface CompareResponse {
  schema_version: string
  items: CompareItem[]
}

export async function fetchJobs() {
  return requestJson<Job[]>('/jobs')
}

export async function createJob(payload: CreateJobPayload) {
  return requestJson<Job>('/jobs', {
    method: 'POST',
    body: JSON.stringify(payload),
  })
}

export async function fetchJob(id: string) {
  return requestJson<Job>(`/jobs/${id}`)
}

export async function compareJobs(jobIds: number[]) {
  return requestJson<CompareResponse>('/jobs/compare', {
    method: 'POST',
    body: JSON.stringify({ job_ids: jobIds }),
  })
}

File: backend/app/schemas/jobs.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class JobCreate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    raw_text: str = Field(min_length=20)


class JobRead(BaseModel):
    id: int
    title: str
    raw_text: str
    profile: dict
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class JobCompareRequest(BaseModel):
    job_ids: list[int] = Field(min_length=2, max_length=5)


class JobCompareDimension(BaseModel):
    name: str
    category: str
    required_level: str
    weight: float


class JobCompareItem(BaseModel):
    job_id: int
    job_title: str
    dimensions: list[JobCompareDimension] = Field(default_factory=list)


class JobCompareResponse(BaseModel):
    schema_version: str = "1"
    items: list[JobCompareItem]
